### PR TITLE
Feature - Get user location

### LINKED ISSUE #203: Obtain the geolocation of users
Get geolocation of users and send it to cboard API.

### RESOURCES
based on these Articles:
https://www.softwaretestinghelp.com/best-ip-geolocation-api/
https://stackoverflow.com/a/35123097
Consider the uses of the following resources:

1. https://www.geoplugin.com
prices to get SSL: https://www.geoplugin.com/webservices/ssl

:+1: 

- Free Acces without a token.
- High precision of city.

:-1: 

- No SSL connection without a paid account

2. https://geolocation-db.com

:+1: 

- Free Acces without a token.
- SSL connection without a paid account

:-1: 

- Low precision of city.

3. https://ipdata.co/
prices: https://ipdata.co/pricing.html

:+1: 

- 1500 free requests daily with a token. (should be stored on API)
- SSL connection with a free account
- High precision of city.

:-1: 

- is necessary a token that should be stored on our API.

5. https://ip-api.com/
Commercial use prices https://members.ip-api.com/#pricing

:+1: 

- 1500 free requests daily with a token. (should be stored on API)
- SSL connection with a free account
- High precision of city.

:-1: 
- Non-commercial use for free.
- No SSL connection without a paid account.

### When should store data?
-Should store the location when a Cboard user creates a new account.
-In the case that users that already have accounts should evaluate when make the requests.
       Consider checking if the location is available at a login moment. Compare if it is available to make the request or not

### PR DESCRIPTION
**Pull request by RodriSanchez1**
close #203

### CODE REVIEW

File: api/controllers/user.js [hunk 1]
```diff
@@ -319,12 +342,59 @@ function loginUser(req, res) {
...
 
+async function updateUserLocation(ip, user) {
+  if (!user.location || !user.location.country) {
+    const newLocation = await getLocation(ip);
+    if (newLocation && newLocation.country) {
+      user.location = newLocation;
+      try {
+        const dbUser = await user.save();
+        if (!dbUser) {
+          user.location = {};
```
Comment (tomivm) [inline api/controllers/user.js:358]: i think that is better to set `user.location = null`
for me, it is more clear that the location is not set

File: api/controllers/user.js [hunk 2]
```diff
@@ -319,12 +342,59 @@ function loginUser(req, res) {
...
+          console.log("Unable to find user on the DB")
+          return;
+        }
+      }
+      catch (err) {
+        console.log("Error saving user location", err)
+        user.location = {};
```
Comment (tomivm) [inline api/controllers/user.js:365]: The same as the comment before
https://github.com/cboard-org/cboard-api/pull/204/files#r927164259

File: api/controllers/user.js [hunk 3]
```diff
@@ -319,12 +342,59 @@ function loginUser(req, res) {
...
       };
-
       return res.status(200).json(response);
     }
   });
 }
 
+async function updateUserLocation(ip, user) {
+  if (!user.location || !user.location.country) {
+    const newLocation = await getLocation(ip);
```
Comment (martinbedouret) [inline api/controllers/user.js:352]: try/catch here

File: api/controllers/user.js [hunk 4]
```diff
@@ -319,12 +342,59 @@ function loginUser(req, res) {
...
+  }
+}
+
+async function getLocation(ip) {
+  return new Promise((resolve, reject) => {
+    ipinfo.lookupIp(ip).then(
+      res => {
+        if (!res.country)
+          throw Error("The location retrieved from Ip Info was incorrect")
+
```
Comment (martinbedouret) [inline api/controllers/user.js:378]: use reject

**Review by martinbedouret — CHANGES_REQUESTED** [review]

**Pull request merged**